node/connection_base.js

```diff
@@ -197,7 +197,11 @@ TChannelConnectionBase.prototype.resetAll = function resetAll(err) {
     outOpKeys.forEach(function eachOutOp(id) {
         var req = self.requests.out[id];
         delete self.requests.out[id];
-        // TODO: shared mutable object... use Object.create(err)?
+        
+        err = errors.TChannelConnectionResetError(err, {
+            connection: self
```

Comment:
Make sense. I will remove connection field, since the error is very much self explanatory.